controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -1515,3 +1443,153 @@ def test_identity_probe(self):
+
+        self._assertResponse(response, grpc.StatusCode.INVALID_ARGUMENT, "volume id")
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_validate_volume_cap_with_wrong_secrets(self, storage_agent):
+        self._test_request_with_wrong_secrets(storage_agent)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_validate_volume_cap_with_wrong_parameters(self, storage_agent):
+        self._test_request_with_wrong_parameters(storage_agent)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_validate_volume_cap_with_unsupported_access_mode(self, storage_agent):
+        storage_agent.return_value = self.storage_agent
+        self.request.volume_capabilities[0].access_mode.mode = 999
+
+        response = self.servicer.ValidateVolumeCapabilities(self.request, self.context)
+
+        self._assertResponse(response, grpc.StatusCode.INVALID_ARGUMENT, "unsupported access mode")
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_validate_volume_cap_with_unsupported_fs_type(self, storage_agent):
+        storage_agent.return_value = self.storage_agent
+
+        volume_capability = utils.get_mock_volume_capability(fs_type="ext3")
+        self.request.volume_capabilities = [volume_capability]
+
+        response = self.servicer.ValidateVolumeCapabilities(self.request, self.context)
+
+        self._assertResponse(response, grpc.StatusCode.INVALID_ARGUMENT, "fs_type")
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_validate_volume_cap_with_no_capabilities(self, storage_agent):
+        storage_agent.return_value = self.storage_agent
+        self.request.volume_capabilities = {}
+
+        response = self.servicer.ValidateVolumeCapabilities(self.request, self.context)
+
+        self._assertResponse(response, grpc.StatusCode.INVALID_ARGUMENT, "not set")
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_validate_volume_cap_with_bad_id(self, storage_agent):
+        storage_agent.return_value = self.storage_agent
+        self.request.volume_id = "wwn1"
+
+        response = self.servicer.ValidateVolumeCapabilities(self.request, self.context)
+
+        self._assertResponse(response, grpc.StatusCode.NOT_FOUND, "id format")
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_validate_volume_cap_with_vol_not_found(self, storage_agent):
+        storage_agent.return_value = self.storage_agent
+        self.mediator.get_object_by_id.return_value = None
+
+        response = self.servicer.ValidateVolumeCapabilities(self.request, self.context)
+
+        self._assertResponse(response, grpc.StatusCode.NOT_FOUND, "wwn")
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_validation_with_vol_context_not_match(self, storage_agent):

```

Comment:
please keep the `test_validate_volume_cap_` prefix for all `ValidateVolumeCapabilities` tests

Reply:
done